web/src/components/Upload/util.ts

```diff
@@ -1,43 +1,115 @@
+  if (textLines.length > 7) {
+    const startTime: string = textLines[6].split(',')[1]
+    const startDate: string = textLines[7].split(',')[1]
+    const startDatetime: moment.Moment = moment.tz(
+      `${startDate} ${startTime}`,
+      'MM-DD-YYYY hh:mm:ss a',
+      'America/Los_Angeles'
+    )
+    return startDatetime
+  } else {
+    return moment('')
+  }
+}
 
-const parseDateFromLogFileName = (fileName: string): string => {
-  const fileNameArr = fileName.split('_')
-  const dateString = fileNameArr[1]
-  return `${dateString.slice(0, 4)}/${dateString.slice(
-    4,
-    6
-  )}/${dateString.slice(6)}`
+export const getDustrakEnd = (
+  textLines: Array<string>,
+  startDatetime: moment.Moment
+): moment.Moment => {
+  if (textLines.length > 8) {
+    const endDatetime: moment.Moment = startDatetime.clone()
+    const testLength: string = textLines[8].split(',')[1]
+    const [days, hours, minutes]: Array<string> = testLength.split(':')
+    endDatetime
+      .add(days, 'days')
+      .add(hours, 'hours')
+      .add(minutes, 'minutes')
+    return endDatetime
+  } else {
+    return moment('')
+  }
 }
 
-export const validateFiles = (files: Array<FileWithPath>): string => {
+export const identFiles = (
+  files: Array<FileWithPath>
+): Array<File | undefined> => {
+  let logFile: File | undefined
+  let csvFile: File | undefined
+
+  files.forEach(file => {
+    if (file.name.endsWith('.csv')) {
+      csvFile = file
+    }
+
+    if (file.name.endsWith('.log')) {
+      logFile = file
+    }
+  })
+  return [logFile, csvFile]
+}
+
+export const validateFiles = async (
+  files: Array<FileWithPath>
+): Promise<string> => {
   if (files.length > 0 && files.length < 2) {
     return 'We need one GPS log file and one DusTrak cvs file. Please add a file to continue.'
   }
-
   if (files.length > 2) {
     return 'We need exactly one GPS log file and one DusTrak cvs file. Please remove additional files.'
   }
-
   if (files.length === 2) {
-    let logFile: FileWithPath | undefined
-    let csvFile: FileWithPath | undefined
-
-    files.forEach(file => {
-      if (file.name.endsWith('.csv')) {
-        csvFile = file
-      }
-
-      if (file.name.endsWith('.log')) {
-        logFile = file
-      }
-    })
-
+    const [logFile, csvFile] = identFiles(files)
     if (!logFile || !csvFile) {
       return 'We need one GPS log file and one DusTrak csv file. Please replace one of your files to continue.'
     }
-  }
 
+    const [logText, csvText]: Array<string> = await Promise.all([
+      logFile.text(),
+      csvFile.text()
+    ])
+    const csvTextLines: Array<string> = csvText.split('\n', 9)
+    const dustrakStart: moment.Moment = getDustrakStart(csvTextLines)
+    const dustrakEnd: moment.Moment = getDustrakEnd(csvTextLines, dustrakStart)
+    const logTextLines: Array<string> = logText.split('\n', 10)
+    const gpsStart: moment.Moment = getGpsStart(logTextLines)
+
+    if (
+      !dustrakStart.isValid() ||
```

Comment:
to clarify, is the empty `moment('')` invalid? Just wanted to be sure that doesn't return the current date instead since I think `moment()` without anything passed into it does that.

Reply:
Correct